### PR TITLE
Optional api2 logs filtering [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by zxan1285**

### CODE REVIEW

**Comment by auth0extensions** [comment]
MD5: 7319468847d7b1aee40dbf5dd963c999

# NSP - Security Report

_This is an automated security audit of this project. Please do not modify its contents_

Vulnerabilities found: **6**
Affected Modules: **superagent, hoek, superagent, mime, mime, superagent**
Summary:

- Large gzip Denial of Service
- Prototype pollution attack
- Large gzip Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Large gzip Denial of Service


## Detailed report
#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-log-extension-tools@1.3.1 > auth0-extension-tools@1.3.1 > superagent@2.3.0**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-log-extension-tools@1.3.1 > auth0-extension-tools@1.3.1 > jsonwebtoken@7.4.3 > joi@6.10.1 > hoek@2.16.3**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-log-extension-tools@1.3.1 > auth0-extension-tools@1.3.1 > webtask-tools@2.2.0 > superagent@1.8.5**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-log-extension-tools@1.3.1 > auth0-extension-tools@1.3.1 > webtask-tools@2.2.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-log-extension-tools@1.3.1 > superagent@1.2.0 > mime@1.3.4**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.2.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-log-extension-tools@1.3.1 > superagent@1.2.0**

---

File: README.md [hunk 1]
```diff
@@ -9,7 +9,8 @@ const client = new LogsApiClient({
   domain: config('AUTH0_DOMAIN'),
   clientId: config('AUTH0_CLIENT_ID'),
   clientSecret: config('AUTH0_CLIENT_SECRET'),
-  tokenCache: myTokenCache
+  tokenCache: myTokenCache,
+  apiFiltering: config('AUTH0_RTA').replace('https://', '') === 'auth0.auth0.com'
```
Comment (fyockm) [inline README.md]: does this belong in the README?

Comment (zxan1285) [reply README.md]: This is possible usage of new option. AFAIK we cannot use api-side filtering on appliances. So for appliances it should be `false`. Unless this has been changed lately, and now we can query filtered logs on appliances.

Comment (fyockm) [reply README.md]: I wasn't questioning the logic, just if it belonged in the README. fine to keep it.

**Pull request closed without merging**